File: IT/Multithreading/C++/multithreading1.cpp
/*
file multithreading1.cpp
compilare con: g++ -std=c++11 multithreading1.cpp -lpthread
*/

#include <iostream>
#include <chrono>
#include <thread>

using std::cout;
using std::endl;
using std::chrono::seconds;
using std::this_thread::sleep_for;

int main()
{
	cout << "Thread main" << endl;

	// arresta il processo principale per 5 secondi
	sleep_for(seconds(5));

	cout << "Fine del thread main." << endl;
}

File: IT/2/Multithreading/Java/Multithreading1.java
/* file Multithreading1.java */

public class Multithreading1 {
	public static void main(String [] args) {
		try
		{
			// ottiene un riferimento al thread principale dell'applicazione
			Thread main = Thread.currentThread();
			String name = main.getName();
			System.out.println("Thread " + name);

			// arresta il processo principale per 5 secondi
			main.sleep(5000);

			System.out.println("Fine del thread " + Thread.currentThread().getName() + ".");
		}
		catch (InterruptedException e) {
			System.err.println(e);
		}
	}
}

